src/config/src/config.rs

```diff
@@ -288,6 +288,37 @@ pub static SMTP_CLIENT: Lazy<Option<AsyncSmtpTransport<Tokio1Executor>>> = Lazy:
     }
 });
 
+pub static SNS_CLIENT: Lazy<aws_sdk_sns::Client> = Lazy::new(|| {
+    let cfg = get_config();
+
+    let region = aws_config::Region::new(cfg.sns.region_name.clone());
+    let mut config_builder = aws_sdk_sns::config::Builder::new()
+        .region(region)
+        .endpoint_url(cfg.sns.endpoint.clone());
+
+    if !cfg.sns.use_default_credentials {
+        let credentials = aws_sdk_sns::config::Credentials::new(
+            cfg.sns.access_key.clone(),
```

Comment:
Use the AWS default credential chain instead of providing credentials using specific environment variables.

Reply:
That was indeed my first instinct but since we were setting config for S3 via specific env variables 

https://github.com/openobserve/openobserve/blob/f0cd9fd3b10214b5d8df26f4e265c5302ee5157f/src/config/src/config.rs#L1143-L1146

and 

https://github.com/openobserve/openobserve/blob/f0cd9fd3b10214b5d8df26f4e265c5302ee5157f/src/infra/src/storage/remote.rs#L254-L259

I thought this approach would be more consistent with the existing code. Although I do believe using default credential chains is definitely the way to go.

Because processing the chain requires several steps and it being an async operation, we’ll can’t simply use LazyStatic, we’ll need to initialize the client like how we load chrome launcher configs

https://github.com/openobserve/openobserve/blob/f0cd9fd3b10214b5d8df26f4e265c5302ee5157f/src/config/src/config.rs#L219-L260

I’ve pushed new changes upstream, please feel free to let me know your thoughts or any other changes we should look into.

Reply:
We will need a different strategy for credentials for s3 and SNS. s3 mechanism needs to support GCS, MinIO, and dozens more compatible object stores and not just s3. In contrast SNS is not offered by any provider other than AWS.

Reply:
Ah that makes sense and perhaps why we have `provider` field 

https://github.com/openobserve/openobserve/blob/f0cd9fd3b10214b5d8df26f4e265c5302ee5157f/src/config/src/config.rs#L1137-L1138

used via

https://github.com/openobserve/openobserve/blob/f0cd9fd3b10214b5d8df26f4e265c5302ee5157f/src/infra/src/storage/remote.rs#L304-L323

Thank you for the clarification.

So essentially the distinction now:

- For SNS - AWS default credential chain as it's specific to AWS.
- For S3 - maintain the current approach to support S3-compatible services beyond just AWS.